### PR TITLE
feat: implements template changes for REST server-streaming.

### PR DESCRIPTION
**Pull request by atulep**
Implements template changes for REST server-streaming.

To do that, `rest` templates were patched to invoke `gapic-common`'s `ServerStream` library from the `service_stub`. To test efficacy, new unit and integration tests were generated for `showcase`.

### CODE REVIEW

File: gapic-generator/lib/gapic/presenters/method_rest_presenter.rb [hunk 1]
```diff
@@ -116,10 +116,19 @@ def nonstandard_lro?
...
+        (@main_method.kind == :normal || @main_method.kind == :server) &&
           http_bindings.first&.verb? &&
           http_bindings.first&.path?
       end
+
+      ##
+      # Whether this method is a server-streaming method
+      #
+      # @return [Boolean]
+      #
```
Comment (atulep) [inline gapic-generator/lib/gapic/presenters/method_rest_presenter.rb:128]: Remove

Comment (atulep) [reply gapic-generator/lib/gapic/presenters/method_rest_presenter.rb:128]: Actually, we need it. In `gapic-generator/templates/default/service/rest/client/method/def/_response.erb`, it checks `<%- elsif method.rest.server_streaming? -%>`. I could remove `.rest`, but all other `if` conditions use it, so prefer to leave it as is.

File: gapic-generator/templates/default/service/rest/service_stub/method/_def.erb [hunk 2]
```diff
@@ -7,13 +7,23 @@
 # @param options [::Gapic::CallOptions]
 #   Overrides the default settings for this call, e.g, timeout, retries etc. Optional.
 #
+<%- if method.server_streaming? -%>
+# @yieldparam chunk [String] The chunk of data received during server streaming.
+#
+# @return [Faraday::Response]
```
Comment (dazuma) [inline gapic-generator/templates/default/service/rest/service_stub/method/_def.erb:13]: Please use `::Faraday::Response` (including leading colons)

File: gapic-generator/templates/default/service/rest/service_stub/method/_def.erb [hunk 3]
```diff
@@ -7,13 +7,23 @@
 # @param options [::Gapic::CallOptions]
 #   Overrides the default settings for this call, e.g, timeout, retries etc. Optional.
 #
+<%- if method.server_streaming? -%>
+# @yieldparam chunk [String] The chunk of data received during server streaming.
```
Comment (dazuma) [inline gapic-generator/templates/default/service/rest/service_stub/method/_def.erb:11]: Please use `::String` (including leading colons)

File: gapic-generator/templates/default/service/rest/client/method/docs/_result.erb [hunk 4]
```diff
@@ -1,6 +1,10 @@
 <%- assert_locals method -%>
+<%- if method.server_streaming? -%>
+# @return [Enumerable<<%= method.rest.doc_response_type %>>]
```
Comment (dazuma) [inline gapic-generator/templates/default/service/rest/client/method/docs/_result.erb:3]: Please use `::Enumerable` (including leading colons)

File: shared/test/showcase/echo/expand_test.rb [hunk 5]
```diff
@@ -52,4 +59,73 @@ def test_expand_with_metadata
...
+  # {"error":{"code":500,"message":"my-error-message","details":[],"Body":"","Header":null,"Errors":null}}UNNECESSARY"
+  # }]
+  # --------------------------------------------------
+  #
+  def test_expand_with_error
+    ret = nil
+    a = []
+    ex = assert_raises Gapic::Common::Error do
+      ret = @client.expand({content: "Sphinx of black quartz, judge my vow.", 
+                                 error: {code: 404, message: "my-error-message" }})
```
Comment (dazuma) [inline shared/test/showcase/echo/expand_test.rb:123]: Please fix indentation

**Review by dazuma — CHANGES_REQUESTED** [review]

File: shared/test/showcase/echo/expand_test.rb (hunk 5 above)
Comment (atulep) [reply shared/test/showcase/echo/expand_test.rb]: Done

File: gapic-generator/templates/default/service/rest/client/method/docs/_result.erb (hunk 4 above)
Comment (atulep) [reply gapic-generator/templates/default/service/rest/client/method/docs/_result.erb]: Done

File: gapic-generator/templates/default/service/rest/service_stub/method/_def.erb (hunk 3 above)
Comment (atulep) [reply gapic-generator/templates/default/service/rest/service_stub/method/_def.erb]: Done

File: gapic-generator/templates/default/service/rest/service_stub/method/_def.erb (hunk 2 above)
Comment (atulep) [reply gapic-generator/templates/default/service/rest/service_stub/method/_def.erb]: Done

File: gapic-generator/lib/gapic/presenters/method_rest_presenter.rb [hunk 6]
```diff
@@ -116,10 +116,19 @@ def nonstandard_lro?
       # @return [Boolean]
       #
       def can_generate_rest?
-        (@main_method.kind == :normal) &&
+        (@main_method.kind == :normal || @main_method.kind == :server) &&
```
[the remark this answers is not in the document]
Comment (atulep) [reply gapic-generator/lib/gapic/presenters/method_rest_presenter.rb]: Done

**Review by dazuma — APPROVED** [review]
I think you need to regenerate the goldens, but otherwise lgtm.

**Pull request merged**